### PR TITLE
Fastjson2 Support

### PR DESCRIPTION
**Pull request by AlbumenJ**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-serialization/dubbo-serialization-fastjson2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-serialization-api</artifactId>
+            <version>${project.parent.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.alibaba.fastjson2</groupId>
+            <artifactId>fastjson2</artifactId>
+            <version>2.0.7</version>
```
Comment (CrazyHZM) [inline dubbo-serialization/dubbo-serialization-fastjson2/pom.xml:46]: External dependencies and version information can be written in `dubbo-dependencies-bom.pom`

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/Constants.java [hunk 2]
```diff
@@ -79,7 +79,11 @@ public interface Constants {
 
     String SERIALIZATION_KEY = "serialization";
 
-    String DEFAULT_REMOTING_SERIALIZATION = "hessian2";
+    String DEFAULT_REMOTING_SERIALIZATION_PROPERTY_KEY = "DUBBO_DEFAULT_SERIALIZATION";
+
+    String DEFAULT_REMOTING_SERIALIZATION_PROPERTY = "hessian2";
+
+    String DEFAULT_REMOTING_SERIALIZATION = System.getProperty(DEFAULT_REMOTING_SERIALIZATION_PROPERTY_KEY, DEFAULT_REMOTING_SERIALIZATION_PROPERTY);
```
[the remark this answers is not in the document]
Comment (AlbumenJ) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/Constants.java:86]: Yep, for migration purpose.

File: dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/support/DefaultSerializationSelector.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            String fromEnv = System.getenv(DEFAULT_REMOTING_SERIALIZATION_PROPERTY_KEY);
+            if (fromEnv != null) {
+                DEFAULT_REMOTING_SERIALIZATION = fromEnv;
+            } else {
+                DEFAULT_REMOTING_SERIALIZATION = DEFAULT_REMOTING_SERIALIZATION_PROPERTY;
+            }
+        }
+    }
+
+    public static String getDefaultRemotingSerialization() {
```
[the remark this answers is not in the document]
Comment (AlbumenJ) [reply dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/support/DefaultSerializationSelector.java:42]: `Constants.DEFAULT_REMOTING_SERIALIZATION` should be removed

**Review by CrazyHZM — APPROVED** [review]
LGTM

**Pull request merged**